### PR TITLE
Add test case test_csi_snapshot_snap_delete_csi_snapshot

### PR DESCRIPTION
**Pull request by chriscchien**
Add test case test_csi_snapshot_snap_delete_csi_snapshot

Ref [4995](https://github.com/longhorn/longhorn/issues/4995)

Signed-off-by: Chris Chien <chris.chien@suse.com>

### CODE REVIEW

File: manager/integration/tests/test_csi_snapshotter.py [hunk 1]
```diff
@@ -983,45 +1013,71 @@ def test_csi_snapshot_snap_delete_csi_snapshot():
...
-    pass
+    # Step 0
+    vol, deployment, csisnapclass, expected_md5sum = \
+        prepare_test_csi_snapshot(
+                                  apps_api, # NOQA
+                                  client, # NOQA
+                                  make_deployment_with_pvc, # NOQA
+                                  volumesnapshotclass, # NOQA
+                                  core_api # NOQA
+                                )
```
Comment (c3y1huang) [inline manager/integration/tests/test_csi_snapshotter.py:1024]: Indent looks weird.

File: manager/integration/tests/test_csi_snapshotter.py [hunk 2]
```diff
@@ -269,6 +270,31 @@ def get_volumesnapshotcontent(volumesnapshot_uid):
...
+                                             version=api_version,
+                                             namespace=namespace,
+                                             plural=plural,
+                                             name=name)
+        except Exception:
+            deleted = True
+            break
+        time.sleep(RETRY_INTERVAL)
+
+    return deleted
```
Comment (c3y1huang) [inline manager/integration/tests/test_csi_snapshotter.py:295]: Why not assert deleted?

**Review by c3y1huang — CHANGES_REQUESTED** [review]

File: manager/integration/tests/test_csi_snapshotter.py (hunk 2 above)
Comment (chriscchien) [reply manager/integration/tests/test_csi_snapshotter.py]: Updated`

File: manager/integration/tests/test_csi_snapshotter.py [hunk 3]
```diff
@@ -947,13 +974,17 @@ def test_csi_snapshot_snap_create_volume_from_snapshot(apps_api, # NOQA
...
 
 
-@pytest.mark.skip(reason="TODO") # NOQA
-def test_csi_snapshot_snap_delete_csi_snapshot():
+def test_csi_snapshot_snap_delete_csi_snapshot(apps_api, # NOQA
+                                      client, # NOQA
+                                      make_deployment_with_pvc, # NOQA
+                                      volumesnapshotclass, # NOQA
+                                      volumesnapshot, # NOQA
+                                      core_api): # NOQA
```
Comment (c3y1huang) [inline manager/integration/tests/test_csi_snapshotter.py:982]: Can you also fix the indents here?

**Review by c3y1huang — COMMENTED** [review]
It seems there are 3 cases here. Suggest separating them.

File: manager/integration/tests/test_csi_snapshotter.py (hunk 3 above)
Comment (chriscchien) [reply manager/integration/tests/test_csi_snapshotter.py]: Updated.

**Review by c3y1huang — CHANGES_REQUESTED** [review]
> Add test

Suggest updating the commit title to something more descriptive, ex: `Implement test_csi_snapshot_snap_delete_csi_snapshot cases`

**Comment by chriscchien** [comment]
> > Add test
> 
> Suggest updating the commit title to something more descriptive, ex: `Implement test_csi_snapshot_snap_delete_csi_snapshot cases`

Updated!

**Review by c3y1huang — APPROVED** [review]

**Comment by yangchiu** [comment]
@Mergifyio backport v1.4.x

**Pull request merged**